src/containers/forms/PractitionerForm/UserIdSelect/index.tsx

```diff
@@ -102,10 +102,18 @@ export const UserIdSelect: React.FC<Props> = props => {
     setOpenMRSUsers(unMatchedUsers);
   };
 
-  const options = openMRSUsers.map((user: OpenMRSUser) => ({
-    label: user.display,
-    value: user.uuid,
-  }));
+  const options = React.useMemo(() => {
```

Comment:
Nice! It would be cool to measure the performance boost here, if it's substantial using React Memos could be pretty useful when dealing with the heavy IRS location hierarchies.